runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs

```diff
@@ -385,13 +387,17 @@ fn get_snapshot_file_kind(filename: &str) -> Option<SnapshotFileKind> {
 }
 
 /// Get the slot and append vec id from the filename
-fn get_slot_and_append_vec_id(filename: &str) -> (Slot, usize) {
-    let mut split = filename.split('.');
-    let slot = split.next().unwrap().parse().unwrap();
-    let append_vec_id = split.next().unwrap().parse().unwrap();
-    assert!(split.next().is_none());
-
-    (slot, append_vec_id)
+pub(crate) fn get_slot_and_append_vec_id(filename: &str) -> (Slot, usize) {
+    STORAGE_FILE_REGEX
+        .captures(filename)
+        .map(|cap| {
+            let slot_str = cap.name("slot").map(|m| m.as_str());
+            let id_str = cap.name("id").map(|m| m.as_str());
+            let slot: Slot = slot_str.unwrap().parse::<u64>().unwrap();
+            let id = id_str.unwrap().parse::<u32>().unwrap() as usize;
```

Comment:
```suggestion
            let slot_str = cap.name("slot").map(|m| m.as_str()).unwrap();
            let id_str = cap.name("id").map(|m| m.as_str()).unwrap();
            let slot = slot_str.parse().unwrap();
            let id = id_str.parse().unwrap();
```

* Move unwraps for regex captures into the assignments for *_str variables. This makes it so those variables are actually `&str`, and will make it more clear the regex mismatched instead of parsing in case of error.
* Also we don't need the type annotations or parse type specifications, they're all implied by return type of the function.